### ISSUE TITLE
Use built-in functions in a custom Engine, without instantiating RecalcEngine

### ISSUE THREAD

**Issue by susosuso** [microsoft/Power-Fx#799]
We want to create our own custom `Engine` instead of using `RecalcEngine`, but we'd still like to use some of the built-in functions provided in the internal library:

https://github.com/microsoft/Power-Fx/blob/2a82bc542d94d9ffd9e5bc81ce46c30dca53e8dd/src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs#L51

Can we provide something like `ReadOnlySymbolTable.NewDefault()` to allow us to obtain the functions without creating RecalcEngine as a throwaway, intermediate resource? 

Or even better, perhaps something like `SymbolTable.NewDefault()`, so we can modify the set without having to call `GetMutableCopyOfFunctions` first to selectively remove the ones we don't want.

If there's a problem with it because SymbolTable is in Core and the library is in Interpreter... then maybe a `CreateNewDefaultSymbolTable()` in `PowerFxConfigExtensions.cs`

 Thanks :)

**Comment by MikeStall** [comment]
Currently, you need to do something like this:
`var mylist = new RecalcEngine().SupportedFunctions.GetMutableCopy();`

One challenge is that TexlFunction is still internal, so even if NewDefault was public, what would you pass to it?

**Comment by susosuso** [comment]
Yeah I understand we can just do `var mylist = new RecalcEngine().SupportedFunctions.GetMutableCopy();`. It just seems unnecessary to create an Engine just to get functions from it.

The method I'm proposing doesn't take in paramaters. The name "default" should imply that it loads the default set of built-in functions. Something like this:

```csharp

public SymbolTable CreateNewDefaultSymbolTable() {
        var coreFunctions = Library.FunctionList;
        var s = new SymbolTable
        {
            EnumStoreBuilder = new EnumStoreBuilder(),
            DebugName = $"BuiltinFunctions ({coreFunctions.Count()})"
        };

        foreach (var func in coreFunctions)
        {
            s.AddFunction(func);
        }

        return s;
}

```

**Comment by susosuso** [comment]
Or even...

```csharp
public static void AddBuiltInSimpleFunctions(this SymbolTable symbolTable) {
        var coreFunctions = Library.SimpleFunctionList; // Get from SimpleFunctionImplementations 
        foreach (var func in coreFunctions)
        {
            symbolTable.AddFunction(func);
        }
}
```
It'll be nice for us because we don't want tabular overloads, at least not yet.

And then if we decide to enable overloads for tables, we can do this 
```csharp
        public static void AddSimpleFunctionTabularOverloadFunctions(this SymbolTable symbolTable)
        { 
             var coreFunctions = Library.SimpleFunctionabularOverloadList; // Get from SimpleFunctionTabularOverloadImplementations  
             foreach (var func in coreFunctions)
             {
                // maybe some sanity check here to skip adding the overload if the simpleFunction peer doesn't exist in here.
                symbolTable.AddFunction(func);
             }
        }
```
